app/src/main/java/com/rkm/tasky/network/repository/implementation/AuthorizationRepositoryImpl.kt

```diff
@@ -0,0 +1,48 @@
+package com.rkm.tasky.network.repository.implementation
+
+import com.rkm.tasky.di.IoDispatcher
+import com.rkm.tasky.network.datasource.TaskyRemoteDataSource
+import com.rkm.tasky.network.model.dto.AccessTokenDTO
+import com.rkm.tasky.network.model.request.AccessTokenRequest
+import com.rkm.tasky.network.model.response.asAccessTokenDTO
+import com.rkm.tasky.network.repository.abstraction.AuthorizationRepository
+import com.rkm.tasky.network.util.NetworkError
+import com.rkm.tasky.network.util.safeCall
+import com.rkm.tasky.util.result.EmptyResult
+import com.rkm.tasky.util.result.Result
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class AuthorizationRepositoryImpl @Inject constructor(
+    private val dataSource: TaskyRemoteDataSource,
+    @IoDispatcher private val dispatcher: CoroutineDispatcher
+): AuthorizationRepository {
+    override suspend fun getNewAccessToken(
+        refreshToken: String,
+        userId: String
+    ): Result<AccessTokenDTO, NetworkError.APIError> = withContext(dispatcher) {
+        val result = safeCall {
+            dataSource.getNewAccessToken(
+                AccessTokenRequest(
+                    refreshToken = refreshToken,
+                    userId = userId
+                )
+            )
+        }
+        return@withContext when (result) {
+            is Result.Success -> Result.Success(result.data.asAccessTokenDTO())
+            is Result.Error -> result
+        }
```

Comment:
Assuming that `safeCall` is an extension function on an http client, wouldn’t it make more sense here to have the data source be the one calling `safeCall` instead?

Reply:
I didn't code safe call as an extension function